Feat/download all files

J-Mertl opened the pull request.

Commit:
feat/downloadAllFiles: integrated a DownloadButton

Commit:
feat/downloadAllFiles: bug-fixed - css for DeploymentListEntry; added Tooltip; changed deploy-download (doesn't work yet)

Commit:
feat/downloadAllFiles: fixed deploy-download

Commit:
feat/downloadAllFiles: finished download

lmoesle commented on src/components/Header/DownloadButton.tsx:
```diff
@@ -0,0 +1,26 @@
+                e.stopPropagation();
+                props.onDownloadClick();
+            }}
+            label={t("Download Project")}
```
Du musst noch die Translation anlegen unter public/translations

J-Mertl commented:
fixes made:
- deployment Entries can't move over the timestamp anymore, and instead move into a second line (similar to the bugfix regarding normal Entries)
- added Tooltips to all Headerbuttons by adding it to the class Actionbutton
- fixed the deployment download; up to now downloaded a .html, now the actual file (e.g.: .bpmn)

feature added:
- added a project/repository zip-download button and its functionality in RepositoryFIleSection & RepositorySharedSection; as well as the backend connection in FileUtils

lmoesle reviewed (changes requested):
👍🏻 Gut gemacht. Du musst dir aber nochmal das Thema mit den Translation anschauen, die stimmen glaube ich nicht und ein bisschen was könnte man noch refactoren.

Commit:
feat/downloadAllFiles: implemented feedback
